src/status_im/subs/communities.cljs

```diff
@@ -297,32 +297,62 @@
  (fn [collapsed-categories [_ community-id]]
    (get collapsed-categories community-id)))
 
+(defn- permission-id->permission-value
+  [token-permissions permission-id]
+  (get (into {} token-permissions) permission-id))
+
 (re-frame/reg-sub
  :community/token-gated-overview
  (fn [[_ community-id]]
    [(re-frame/subscribe [:communities/community community-id])])
  (fn [[{:keys [token-permissions-check token-permissions checking-permissions? token-images]}] _]
-   {:can-request-access?   (:satisfied token-permissions-check)
-    :number-of-hold-tokens (reduce
-                            (fn [acc [_ {:keys [criteria]}]]
-                              (reduce #(+ %1 (if %2 1 0)) acc criteria))
-                            0
-                            (:permissions token-permissions-check))
-    :tokens                (->> token-permissions
-                                (filter (fn [[_ {:keys [type]}]]
-                                          (= type constants/community-token-permission-become-member)))
-                                (map (fn [[perm-key {:keys [token_criteria]}]]
-                                       (let [check-criteria (get-in token-permissions-check
-                                                                    [:permissions perm-key :criteria])]
-                                         (map
-                                          (fn [{sym :symbol amount :amount} sufficient?]
-                                            {:symbol      sym
-                                             :sufficient? (when (seq check-criteria) sufficient?)
-                                             :loading?    checking-permissions?
-                                             :amount      amount
-                                             :img-src     (get token-images sym)})
-                                          token_criteria
-                                          (or check-criteria token_criteria))))))}))
+   (let [can-request-access? (:satisfied token-permissions-check)
+         role-permissions #{constants/community-token-permission-become-admin
+                            constants/community-token-permission-become-member
+                            constants/community-token-permission-become-token-master
+                            constants/community-token-permission-become-token-owner}
+         highest-permission-role
+         (when can-request-access?
+           (->> token-permissions-check
```

Comment:
This subscription is turning out to be somewhat complex to understand. I don't know if it's worth refactoring it too much because we can actually consider having a single endpoint in status-go to return all of this, but since we don't have that yet, consider refactoring it into small private functions in this sub namespace, this would go a long of helping understand the code.

Reply:
The permissions and permissions-check in db community are in weird shapes (maybe there's a reason?) and do need some refactors on both mobile and go side

> having a single endpoint in status-go to return all of this

the approach using `wakuext_fetchCommunity` to re-fetch community should be removed and rely on status-go signal status-mobile about community updates
I think permission updates should be handled that way as well (already partially handled? since the video shows mobile react and perm change automatically)

`:communities/check-permissions-to-join-community` (to get token-permissions-check) do many requests to blockchain nodes on every call
which might be a problem if merge it into the same endpoint

maybe balance change -> go signals changes to mobile is a better approach
since after the include all future address feature new wallet account will also trigger changes